### ISSUE TITLE
Cannot search due to old Jquery version being used with "-b readthedocs"

### ISSUE THREAD

**Issue by gerhardol** [rtfd/readthedocs.org#5030]
## Details

* Read the Docs project URL: https://git-extensions-documentation.readthedocs.io/en/master/
* Build URL (if applicable): https://readthedocs.org/projects/git-extensions-documentation/builds/8283885/
* Read the Docs username (if applicable): gerhardol

Documentation cannot be searched: https://github.com/gitextensions/gitextensions/issues/5948

## Expected Result

Search possible and the version switcher popup reacting.

## Actual Result

When loading the page (for instance Inspection in Chrome), the following appears:
readthedocs-doc-embed.js:1 Uncaught TypeError: $(...).on is not a function
    at Object.init (readthedocs-doc-embed.js:1)
    at HTMLDocument.<anonymous> (readthedocs-doc-embed.js:1)
    at Function.ready (jquery.js:26)
    at HTMLDocument.L (jquery.js:33)

The behavior is like JQuery 1.2.1 is actually used and not JQuery 3.2.1 that is bundled: https://git-extensions-documentation.readthedocs.io/en/master/_static/jquery-3.2.1.js

I do not believe this is a Sphinx problem, as building with Sphinx 1.7.9 locally works fine (not in Chrome but Firefox). (no version switcher though.) The readthedocs builds uses "-T -E -b readthedocs" instead of "-b html" 
If I install the readthedocs extension I get this problem too:
pip install https://github.com/rtfd/readthedocs-sphinx-ext/archive/master.zip

The previous build for 2.51 a couple of months ago worked https://git-extensions-documentation.readthedocs.io/en/release-2.51
but unfortunately that was rebuilt. The log from the previous working build looks similar to the new builds though.
Older builds (a year and a half ago) works OK, but they use different versions (JQuery 2.0 for instance) and is hard to compare.

The only error I see in the logfile is 
```copying static files... WARNING: html_static_path entry '/home/docs/checkouts/readthedocs.org/readthedocs/templates/sphinx/_static' does not exist
done```
That part is inserted by readthedocs and I can search locally with a similar change.

**Comment by humitos** [comment]
I suppose that we introduced this bug at #5001 and https://github.com/rtfd/readthedocs-sphinx-ext/pull/52

I don't understand exactly why/how but those PRs touches the jQuery file and I suppose that something went wrong with the symlinks and it ended pointing to an old version.

But it's clear that this is not happening,

> After rtfd/readthedocs-sphinx-ext#52, new builds of Sphinx docs will use the version of jquery bundled with Sphinx. This will remove a source of incompatibility where local Sphinx builds use one version of jquery and RTD Sphinx builds use another.

(from #5001)

> The behavior is like JQuery 1.2.1 is actually used and not JQuery 3.2.1 that is bundled: [git-extensions-documentation.readthedocs.io/en/master/_static/jquery-3.2.1.js](https://git-extensions-documentation.readthedocs.io/en/master/_static/jquery-3.2.1.js)

The version your docs are using is 1.4.2, https://git-extensions-documentation.readthedocs.io/en/master/_static/jquery.js

I didn't find where this file come from :/

```
(docs) docs@web02:~/checkouts/readthedocs.org/user_builds/git-extensions-documentation/rtd-builds/master$ ls -la _static/jquery.js 
-rw-r--r-- 1 docs docs 72174 Dec 16 21:27 _static/jquery.js

(docs) docs@web02:~/checkouts/readthedocs.org/user_builds/git-extensions-documentation/rtd-builds/master$ head _static/jquery.js 
/*!
 * jQuery JavaScript Library v1.4.2
 * http://jquery.com/
 *
 * Copyright 2010, John Resig
 * Dual licensed under the MIT or GPL Version 2 licenses.
 * http://jquery.org/license
 *
 * Includes Sizzle.js
 * http://sizzlejs.com/

(docs) docs@web02:~/checkouts/readthedocs.org/user_builds/git-extensions-documentation/rtd-builds/master$ head _static/jquery-3.2.1.js 
/*!
 * jQuery JavaScript Library v3.2.1
 * https://jquery.com/
 *
 * Includes Sizzle.js
 * https://sizzlejs.com/
 *
 * Copyright JS Foundation and other contributors
 * Released under the MIT license
 * https://jquery.org/license
```

`_static/jquery.js` should be https://github.com/sphinx-doc/sphinx/blob/v1.7.9/sphinx/themes/basic/static/jquery.js in your case.

**Comment by gerhardol** [comment]
The difference between the -b html and -b readthedocs builds is that html target copies the latest versioned files (also included) for jquery and underscore.

Confirming that it is possible to search when overriding the following files (from the versioned files generated by sphinx).

$ ls source/_static/
jquery.js  underscore.js

**Comment by davidfischer** [comment]
It looks like the build used Sphinx 1.7 which should copy jQuery v3.2.1. I don't yet see why it wouldn't copy it. However, I'm guessing this is an old jquery version from a *very* old build of the docs on Read the Docs.

**Comment by davidfischer** [comment]
> However, I'm guessing this is an old jquery version from a _very_ old build of the docs on Read the Docs.

I don't think this is an issue with symlinks because at least some docs are picking up the correct jquery version (eg. [our docs](https://docs.readthedocs.io/en/latest/_static/jquery.js)).

**Comment by gerhardol** [comment]
Both Sphinx -b html and the readthedocs includes:
```
$ ls -l build/html/_static/jquery*
-rw-rw-r--+ 1 ejgo Ingen  72328  2 sep 23.16 build/html/_static/jquery.js
-rw-rw-r--+ 1 ejgo Ingen 268039 22 dec 20.05 build/html/_static/jquery-3.2.1.js
```

Both reference jquery (and underscore) in the html header, without versions.

Both builds fail if the readthedocs extension is loaded

**Comment by gerhardol** [comment]
> but maybe this is the same as

Building with Spinx (1.8.2? and 1.7.1 as used on readthedocsw) (-b html) is OK, but not on readthedocs (-b readthedocs) so it is likely not the problem

**Comment by stsewd** [comment]
I was able to replicate this, just adding more context here:

- If the rtd-ext is in the list of extensions, the search fail (doesn't matter if it's called with `-b html` or `-b readthedocs`).

Can't replicate with a minimal sphinx project, so I guess this is something to do with the theme or another extension. I'm investigating more.

**Comment by stsewd** [comment]
So, I have more info here. Previously we were replacing the jquery file (the theme that you are using uses `1.4.2`) for other https://github.com/rtfd/readthedocs-sphinx-ext/pull/52/files#diff-1b00543baae13c24c6592e94f0ea6c1dL30 (`2.0.3`).

Now, we insert this js file in every build (`html`, `readthedocs`, etc)

https://github.com/rtfd/readthedocs-sphinx-ext/blob/b0048f90aaac4017aa4e0db3ae5e7b5a6a57e841/readthedocs_ext/readthedocs.py#L55-L55

That file makes use of features available in more updated version of jquery (the one that we were replacing before https://github.com/rtfd/readthedocs-sphinx-ext/pull/52/).

So, we could solve this in two ways I think:

- Update the js file from the theme that you are using https://github.com/gitextensions/GitExtensionsDoc/blob/latest/source/nature2/static/jquery.js (looks like a fork from another theme, right? Maybe is worth updating from upstream?)
- Make our js file compatible with lower versions of jquery view-source:https://assets.readthedocs.org/static/javascript/readthedocs-doc-embed.js
- Don't insert that file at all (not sure if we are using it in any project)

**Comment by stsewd** [comment]
We can't remove the file, we are using it to inject the footer https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/core/static-src/core/js/readthedocs-doc-embed.js

**Comment by gerhardol** [comment]
> Maybe is worth updating from upstream?

That seem to work:
https://git-extensions-documentation.readthedocs.io/en/feature-fixes/

So the core problem: Modified theme does not work with "readthedocs_ext.readthedocs" after rtd update even if sphinx "html" is OK.

The theme override could be much simplified after the update, just a few changes to the nature.css_t
A few custom changes was dropped to avoid similar problems. Will maybe look at that later.
(Now just how to move the version selector from right-bottom to left-bottom).

If nothing else comes up, I will close this issue in some days.
